.actions/setup_tools.py

```diff
@@ -42,45 +45,92 @@ def _load_py_module(name: str, location: str) -> ModuleType:
+        ln = ln[: ln.index(comment_char)]
+        is_strict = "strict" in comment
+    else:
+        is_strict = False
+    req = ln.strip()
+    # skip directly installed dependencies
+    if not req or req.startswith("http") or "@http" in req:
+        return ""
+    # extract the major version from all listed versions
+    if unfreeze == "major":
+        req_ = list(parse_requirements([req]))[0]
+        vers = [LooseVersion(v) for s, v in req_.specs if s not in ("==", "~=")]
+        ver_major = sorted(vers)[-1].version[0] if vers else None
+    else:
+        ver_major = None
+
+    # remove version restrictions unless they are strict
+    if unfreeze and "<" in req and not is_strict:
+        req = re.sub(r",? *<=? *[\d\.\*]+", "", req).strip()
+    if ver_major is not None and not is_strict:
+        # add , only if there are already some versions
+        req += f"{',' if any(c in req for c in '<=>') else ''} <{int(ver_major) + 1}.0"
+
+    # adding strict back to the comment
+    if is_strict or ver_major is not None:
+        req += "  # strict"
+
+    return req
+
+
 def load_requirements(
-    path_dir: str, file_name: str = "base.txt", comment_char: str = "#", unfreeze: bool = True
+    path_dir: str, file_name: str = "base.txt", comment_char: str = "#", unfreeze: str = "all"
 ) -> List[str]:
     """Loading requirements from a file.
 
     >>> path_req = os.path.join(_PROJECT_ROOT, "requirements")
-    >>> load_requirements(path_req)  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
-    ['numpy...', 'torch...', ...]
+    >>> load_requirements(path_req, unfreeze="major")  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
+    ['pytorch_lightning...', 'lightning_app...']
     """
     with open(os.path.join(path_dir, file_name)) as file:
         lines = [ln.strip() for ln in file.readlines()]
     reqs = []
     for ln in lines:
-        # filer all comments
-        comment = ""
-        if comment_char in ln:
-            comment = ln[ln.index(comment_char) :]
-            ln = ln[: ln.index(comment_char)]
-        req = ln.strip()
-        # skip directly installed dependencies
-        if not req or req.startswith("http") or "@http" in req:
-            continue
-        # remove version restrictions unless they are strict
-        if unfreeze and "<" in req and "strict" not in comment:
-            req = re.sub(r",? *<=? *[\d\.\*]+", "", req).strip()
-
-        # adding strict back to the comment
-        if "strict" in comment:
-            req += "  # strict"
-
-        reqs.append(req)
-    return reqs
+        reqs.append(_augment_requirement(ln, comment_char=comment_char, unfreeze=unfreeze))
+    # filter empty lines
+    return [str(req) for req in reqs if req]
 
 
 def load_readme_description(path_dir: str, homepage: str, version: str) -> str:
     """Load readme as decribtion.
 
     >>> load_readme_description(_PROJECT_ROOT, "", "")  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
-    '<div align="center">...'
+    '...'
```

Comment:
```suggestion
    '...PyTorch Lightning is just organized PyTorch...'
```

Reply:
Well seems we do not have this tested for a while so let's leave it for another PR as we would also need to edit CI

Reply:
This suggestion is what I did in #13912. The doctest fails because william pushed a commit to master changing the beginning of the readme. My suggestion fixes the doctest

Reply:
addressing in #14498